processor/src/host/advice/mod.rs

```diff
@@ -171,732 +171,668 @@ pub trait AdviceProvider: Sized {
     /// It is not checked whether a Merkle tree for either of the specified roots can be found in
     /// this advice provider.
     fn merge_roots(&mut self, lhs: Word, rhs: Word) -> Result<Word, ExecutionError>;
+}
 
-    // PROVIDED METHODS
-    // --------------------------------------------------------------------------------------------
-
-    // DEFAULT ADVICE MAP INJECTORS
-    // --------------------------------------------------------------------------------------------
-
-    /// Reads words from memory at the specified range and inserts them into the advice map under
-    /// the key `KEY` located at the top of the stack.
-    ///
-    /// Inputs:
-    ///   Operand stack: [KEY, start_addr, end_addr, ...]
-    ///   Advice map: {...}
-    ///
-    /// Outputs:
-    ///   Operand stack: [KEY, start_addr, end_addr, ...]
-    ///   Advice map: {KEY: values}
-    ///
-    /// Where `values` are the elements located in memory[start_addr..end_addr].
-    ///
-    /// # Errors
-    /// Returns an error:
-    /// - `start_addr` is greater than or equal to 2^32.
-    /// - `end_addr` is greater than or equal to 2^32.
-    /// - `start_addr` > `end_addr`.
-    fn insert_mem_values_into_adv_map(
-        &mut self,
-        process: ProcessState,
-    ) -> Result<(), ExecutionError> {
-        let (start_addr, end_addr) = get_mem_addr_range(process, 4, 5)?;
-        let ctx = process.ctx();
-
-        let mut values = Vec::with_capacity(((end_addr - start_addr) as usize) * WORD_SIZE);
-        for addr in start_addr..end_addr {
-            let mem_value = process.get_mem_value(ctx, addr).unwrap_or(EMPTY_WORD);
-            values.extend_from_slice(&mem_value);
-        }
-
-        let key = process.get_stack_word(0);
-        self.insert_into_map(key, values);
+// NATIVE EVENTS
+// ===============================================================================================
```

Comment:
I would probably move the code below into a separate module. Maybe something like `src/host/advice/sys_events.rs`.

Reply:
Moved it to `processor/src/operations/sys_ops/sys_event_handlers.rs`, since this logic is not host-related, but really processor/VM-related (i.e. the VM knows how to use the `Host` to handle the system events)